zos_subsystems/cics/cmci/restart_bundles/restart_bundles.yml

```diff
@@ -0,0 +1,142 @@
+    - name: scope
+      prompt: Target CICSPlex SM scope
+      private: false
+    - name: bundle_name
+      prompt: Name or pattern of the CICS bundles to restart
+      private: false
+    - name: cmci_user
+      prompt: CMCI user name (leave blank for unauthenticated)
+    - name: cmci_password
+      prompt: CMCI password (leave blank for unauthenticated)
+
+  tasks:
+    ############################################################################
+    # Install module dependencies
+    ############################################################################
+    - name: Make sure CMCI module dependencies are installed
+      ansible.builtin.pip:
+        name:
+          - requests
+          - xmltodict
+          - typing;python_version<"3.5"
+
+    - name: Disable CICS bundles
+      ibm.ibm_zos_cics.cmci_update:
+        cmci_host: "{{ cmci_host }}"
+        cmci_port: "{{ cmci_port | int }}"
+        cmci_user: "{{ cmci_user | default(omit) }}"
+        cmci_password: "{{ cmci_password | default(omit) }}"
+        context: "{{ context }}"
+        scope: "{{ scope }}"
+        scheme: "{{ scheme }}"
+        type: "CICSBundle"
+        resources:
+          filter:
+            name: "{{ bundle_name }}"
+        attributes:
+          enablestatus: disabled
+
+
+    - name: Wait for CICS bundles to all be disabled
+      ibm.ibm_zos_cics.cmci_get:
+        cmci_host: "{{ cmci_host }}"
+        cmci_port: "{{ cmci_port | int }}"
+        cmci_user: "{{ cmci_user | default(omit) }}"
+        cmci_password: "{{ cmci_password | default(omit) }}"
+        context: "{{ context }}"
+        scope: "{{ scope }}"
+        scheme: "{{ scheme }}"
+        type: "CICSBundle"
+        resources:
+          filter:
+            name: "{{ bundle_name }}"
+      register: bundlestatus
+      retries: 5
+      until: bundlestatus.records | rejectattr('enablestatus', 'equalto', 'DISABLED') | count == 0
+
+
+    - name: Enable CICS bundles
+      ibm.ibm_zos_cics.cmci_update:
+        cmci_host: "{{ cmci_host }}"
+        cmci_port: "{{ cmci_port | int }}"
+        cmci_user: "{{ cmci_user | default(omit) }}"
+        cmci_password: "{{ cmci_password | default(omit) }}"
+        context: "{{ context }}"
+        scope: "{{ scope }}"
+        scheme: "{{ scheme }}"
+        type: "CICSBundle"
+        resources:
+          filter:
+            name: "{{ bundle_name }}"
+        attributes:
+          enablestatus: enabled
+
+
+    - name: Wait for CICS bundles to enable
+      ibm.ibm_zos_cics.cmci_get:
+        cmci_host: "{{ cmci_host }}"
+        cmci_port: "{{ cmci_port | int }}"
+        cmci_user: "{{ cmci_user | default(omit) }}"
+        cmci_password: "{{ cmci_password | default(omit) }}"
+        context: "{{ context }}"
+        scope: "{{ scope }}"
+        scheme: "{{ scheme }}"
+        type: "CICSBundle"
+        resources:
+          filter:
+            name: "{{ bundle_name }}"
+      register: bundlestatus
+      retries: 5
```

Comment:
I was wondering if we wanted the overall wait to be a bit longer? By default it is 5s, so 25s with 5 retries. Considering this could be a Java bundle or many of them, maybe 10 retries might be more appropriate.